icu4c/source/common/unicode/uobject.h

```diff
@@ -20,29 +20,13 @@
 #define __UOBJECT_H__
 
 #include "unicode/utypes.h"
+#include "unicode/platform.h"
 
 /**
  * \file
  * \brief C++ API: Common ICU base class UObject.
  */
 
-/**
- * \def U_NO_THROW
- *         Define this to define the throw() specification so
- *                 certain functions do not throw any exceptions
- *
- *         UMemory operator new methods should have the throw() specification 
- *         appended to them, so that the compiler adds the additional NULL check 
- *         before calling constructors. Without, if <code>operator new</code> returns NULL the 
- *         constructor is still called, and if the constructor references member 
- *         data, (which it typically does), the result is a segmentation violation.
- *
- * @stable ICU 4.2
- */
-#ifndef U_NO_THROW
```

Comment:
Since this is `@stable`, it can't be deleted.  We can, however, remove all uses, like you did in this PR.

For now, keep this macro, and add a note to the API docs that it is no longer being used by ICU.  Eventually we should deprecate the API.  I filed a follow-up ticket to do this in ICU 65.  You are welcome to take that ticket as well:

https://unicode-org.atlassian.net/browse/ICU-20422

Reply:
Sorry for the delay. Is it Ok like this ?

Reply:
@Romain-Geissler-1A I think @sffc was wanting to not deprecate this time around… can you keep it as `@stable` this time around but just add a note that it "is not used by ICU, see ICU-20422"?

Reply:
Ok, how about this version ?